test/distributed/tensor/test_utils.py

```diff
@@ -19,6 +22,70 @@
 c10d_functional = torch.ops.c10d_functional
 
 
+class LocalTest(TestCase):
+    def test_explicit_order_placements(self):
+        # mesh_shape: ShapeType, placements: Sequence[Placement]
+        test_cases = [
+            {
+                "mesh_shape": [2, 2],
+                "placements": [Replicate()],
+                "ordered": [(0, Replicate())],
+            },
+            {
+                "mesh_shape": [2, 2],
+                "placements": [Shard(0)],
+                "ordered": [(0, Shard(0))],
+            },
+            {
+                "mesh_shape": [2, 2],
+                "placements": [_StridedShard(0, split_factor=2), Shard(0)],
+                "ordered": [(1, Shard(0)), (0, Shard(0))],
+            },
+            {
+                "mesh_shape": [2, 3, 4],
+                "placements": [Shard(0), _StridedShard(0, split_factor=4), Shard(0)],
+                "ordered": [(0, Shard(0)), (2, Shard(0)), (1, Shard(0))],
```

Comment:
@XilunWu now i think this is wrong.  since `(0, Shard(0))` and  `(2, Shard(0))` both happen before the _StridedShard(0), I think the split_factor should be 2 * 4 = 8?  When writing this test I was ignoring that left-hand Shard(0) when considering the strided sharding.

Reply:
ok, yea, i agree.  
Wrong understanding:  Split factor accounts for all of the 'earlier' shardings.  (e.g. dim 0 and then dim 2)
Correct understanding: SF accounts for only the 'earlier' shardings to the right of the SS, since those are the ones we are 'striding' over. (e.g. just dim 2)